### PR TITLE
Front end jazz

### PR DESCRIPTION
**Pull request by ChrisChrisLoLo**
Polish up the nav bar, add a search bar, add a template linking to the user profile page

### CODE REVIEW

**Comment by raysarinas** [comment]
LGTM based on my very basic knowledge of React

**Review by raysarinas — APPROVED** [review]
nice

File: back-end/api/quickstart/test_public_post_list.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  def tests_get_all_public_posts(self):
+    response = client.get(f'/api/public-posts/')
+
+    posts = Post.objects.filter(visibility='Public')
+
+    serializer = PostSerializer(posts, many=True)
+
+    print(response.data)
+    print(serializer.data)
```
Comment (jmmxp) [inline back-end/api/quickstart/test_public_post_list.py:34]: Remove debug prints

File: back-end/api/quickstart/helper_test.py [hunk 2]
```diff
@@ -60,4 +60,27 @@ def get_test_post_fields(i=1, author="testauthorid"):
...
+    'isPrivateToFriends': False,
+    'author': author,
+    'contentType': 'text/plain',
+    'content': 'Hello, I am a test post',
+    'categories': '["Testing"]',
+    'published': '03:25:53.827044',
+    'count': 5,
+    'pageSize': 20,
+    'commentLink': 'link to comments',
+    'comments': '{ "text": "nice test" }'
```
Comment (jmmxp) [inline back-end/api/quickstart/helper_test.py:85]: Should use the get_test_post_fields method but just send in a visibility. I can add this in a future PR.

**Pull request merged**